src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs

```diff
@@ -249,6 +250,14 @@ private bool OnPreprocessor(SyntaxTrivia trivia, int currentIndex)
                     return true;
                 }
 
+                // #if preprocessors can be changed by code actions (eg remove unnecessary parenthesis)
+                // so we need to check for elastic trivia within them in case we're formatting the results
+                if (trivia.Kind() is SyntaxKind.IfDirectiveTrivia or SyntaxKind.ElifDirectiveTrivia)
+                {
+                    var structure = trivia.GetStructure()!;
+                    return structure.DescendantTrivia().Any(t => t.IsElastic());
+                }
```

Comment:
💭 I think I like the general approach here, but would prefer to see this either in `OnElastic` or in a new method `OnStructuredTrivia` that calls `ShouldFormat` on descendant trivia lists.

Reply:
Can you be more specific about which bit you'd want to see?

Checking for these two specific directives in OnElastic felt odd, and I thought that descending into any structured trivia might have unwanted performed implications, but I'm more than happy to do either if you think it's better.

Reply:
Setting the elastic annotation on a trivia within a structured trivia is expected to behave the same as any other elastic trivia. It shouldn't be restricted to just a subset of the syntax kinds.

Reply:
For example:

```
<seecref="object"/>
    ^ Elastic marker here should result in formatting to 'see cref'
```

Reply:
Thanks, updated.